### PR TITLE
Make `stop` wait for the actual termination of the children processes

### PR DESCRIPTION
**Pull request by jfsmig**
Helps fixing #13 

The old-style "fire & forget" `stop`command is still available under the name `kill`.

### CODE REVIEW

**Comment by jfsmig** [comment]
Also, an option has been introduced in `gridinit` named `delay_sigkill` to override the default of 60s before a termination attempt will send SIGKILL instead of SIGTERM.
Set `delay_sigkill` to *0* to never send a SIGTERM.

That variable is available in the sections `[default]` (it changes the global default timeout) or on each `[service.*]` section (to configure one child process at once).

File: lib/children.c [hunk 1]
```diff
@@ -458,14 +458,15 @@ static void
...
 {
 	if (sd->pid > 0) {
-		time_t now = time(0);
+		time_t now = _monotonic_seconds();
 		if (sd->first_kill_attempt == 0)
 			sd->first_kill_attempt = now;
-		if (sd->first_kill_attempt > 0 && (now - sd->first_kill_attempt > SUPERVISOR_DEFAULT_TIMEOUT_KILL)) {
+		if (sd->delay_before_KILL > 0
+                && sd->first_kill_attempt > 0
+                && (now - sd->first_kill_attempt) > sd->delay_before_KILL) {
```
Comment (fvennetier) [inline lib/children.c]: The previous 2 lines are indented with spaces.

File: lib/gridinit-utils.h [hunk 2]
```diff
@@ -73,222 +75,102 @@ typedef void (supervisor_cb_f) (void *udata, struct child_info_s *ci);
...
- */
+/* Stops the UP services that are in state that does not allow them to run.
+   This includes services DOWN, BROKEN, STOPPED, DISABLED.
+   Will send SIGKILL until expiration, then SIGTERM. */
 guint supervisor_children_kill_disabled(void);
 
-/**
- * starts allt the stopped services in a state proper to be restarted
- */
+/* starts allt the stopped services in a state proper to be restarted */
```
Comment (fvennetier) [inline lib/gridinit-utils.h]: *allt* -> *all*

File: main/gridinit_cmd.c [hunk 3]
```diff
@@ -616,13 +578,17 @@ command_reload(int argc, char **args)
...
+struct command_s {
+	const gchar *name;
+	int (*action) (int argc, char **args);
+} COMMANDS[] = {
+	{ "status",  command_status0 },
+	{ "status2", command_status1 },
+	{ "status3", command_status2 },
+	{ "start",   command_start },
+	{ "stop",    command_stop },
+	{ "kill",    command_kill },
```
Comment (fvennetier) [inline main/gridinit_cmd.c:590]: To stay within the spirit, it should have been called "stop2". :wink:

**Pull request merged**